public/app/controllers/chatRoomController.js

```diff
@@ -522,6 +534,13 @@ angular.module('Controllers')
 		chatLog += "\n";
 	});
 
+	$socket.on("announce message", function(data){
```

Comment:
It is recommended that you create a new `Message` instance. It has many useful features.
Also it prevents XSS, since filtering is implemented within `Message` class.

Reply:
Please refer to new message event and use the `Message` class instead.
```
// It should be something like:
let msg = new Chat.Message(data); // Data should be the standard message object
// And you do this instead of directly accessing the property:
msg.getText().getRaw()
```

Reply:
fixed